data_model/src/lib.rs

```diff
@@ -741,6 +741,19 @@ pub trait Registered: Identifiable {
     type With: Into<RegistrableBox>;
 }
 
+/// Trait for proxy objects used for registration.
+pub trait Registrable {
+    /// Constructed type
+    type Target: Identifiable + Registered<With = Self>;
+
+    /// Identification of [`Self::Target`]
+    fn target_id(&self) -> &<Self::Target as Identifiable>::Id;
```

Comment:
this method is redundant. You will never read the `id` of a builder, i.e. `NewXXX` structure. You always know which id you stored into the builder. I gave you a [suggestion](https://github.com/hyperledger/iroha/pull/2503/files#r926764681) in one place that it's used